src/openrct2-ui/windows/Changelog.cpp

```diff
@@ -41,273 +41,308 @@ static constexpr const rct_string_id WINDOW_TITLE = STR_CHANGELOG_TITLE;
+                break;
+            case WIDX_OPEN_URL:
+                if (_newVersionInfo)
+                {
+                    GetContext()->GetUiContext()->OpenURL(_newVersionInfo->url);
+                }
+                else
+                {
+                    log_error("Cannot open URL: NewVersionInfo for ChangelogWindow is undefined!");
+                }
+                break;
+        }
+    }
 
-static void window_changelog_invalidate(rct_window* w)
-{
-    window_changelog_widgets[WIDX_BACKGROUND].right = w->width - 1;
-    window_changelog_widgets[WIDX_BACKGROUND].bottom = w->height - 1;
-    window_changelog_widgets[WIDX_TITLE].right = w->width - 2;
-    window_changelog_widgets[WIDX_CLOSE].left = w->width - 13;
-    window_changelog_widgets[WIDX_CLOSE].right = w->width - 3;
-    window_changelog_widgets[WIDX_CONTENT_PANEL].right = w->width - 1;
-    window_changelog_widgets[WIDX_CONTENT_PANEL].bottom = w->height - 1;
-    window_changelog_widgets[WIDX_SCROLL].right = w->width - 3;
-    window_changelog_widgets[WIDX_SCROLL].bottom = w->height - 22;
-    window_changelog_widgets[WIDX_OPEN_URL].bottom = w->height - 5;
-    window_changelog_widgets[WIDX_OPEN_URL].top = w->height - 19;
-}
+    void OnScrollDraw(int32_t scrollIndex, rct_drawpixelinfo& dpi) override
+    {
+        const int32_t lineHeight = font_get_line_height(FontSpriteBase::MEDIUM);
 
-static void window_changelog_paint(rct_window* w, rct_drawpixelinfo* dpi)
-{
-    WindowDrawWidgets(w, dpi);
-}
+        ScreenCoordsXY screenCoords(3, 3 - lineHeight);
+        for (const auto& line : _changelogLines)
+        {
+            screenCoords.y += lineHeight;
+            if (screenCoords.y + lineHeight < dpi.y || screenCoords.y >= dpi.y + dpi.height)
+                continue;
 
-static void window_changelog_scrollpaint(rct_window* w, rct_drawpixelinfo* dpi, [[maybe_unused]] int32_t scrollIndex)
-{
-    const int32_t lineHeight = font_get_line_height(FontSpriteBase::MEDIUM);
+            gfx_draw_string(&dpi, screenCoords, line.c_str(), { colours[0] });
+        }
+    }
 
-    ScreenCoordsXY screenCoords(3, 3 - lineHeight);
-    for (const auto& line : _changelogLines)
+    ScreenSize OnScrollGetSize(int32_t scrollIndex) override
     {
-        screenCoords.y += lineHeight;
-        if (screenCoords.y + lineHeight < dpi->y || screenCoords.y >= dpi->y + dpi->height)
-            continue;
-
-        gfx_draw_string(dpi, screenCoords, line.c_str(), { w->colours[0] });
+        return ScreenSize(
+            _changelogLongestLineWidth + 4, (int32_t)_changelogLines.size() * font_get_line_height(FontSpriteBase::MEDIUM));
     }
-}
 
-static void window_changelog_process_changelog_text(const std::string& text)
-{
-    std::string::size_type pos = 0;
-    std::string::size_type prev = 0;
-    while ((pos = text.find("\n", prev)) != std::string::npos)
+    // TODO(NBurley93): This probably should be a utility function defined elsewhere for reusability
```

Comment:
TODO is fine but we don't put usernames in the code.

Reply:
Technically in the coding style it says its fine to sign off comments with username. But ive not seen it in any.

Reply:
I'll remove my username for brevity

Reply:
> Technically in the coding style it says its fine to sign off comments with username. But ive not seen it in any.

That is why I thought we don't do it, but given we have none perhaps we should change the style guide.